cmd/server/main.go

```diff
@@ -1,31 +1,93 @@
 package main
 
 import (
+	"fmt"
 	"github.com/DimaKoz/practicummetrics/internal/common/config"
+	"github.com/DimaKoz/practicummetrics/internal/common/repository"
 	"github.com/DimaKoz/practicummetrics/internal/server/handler"
+	middleware2 "github.com/DimaKoz/practicummetrics/internal/server/middleware"
 	"github.com/labstack/echo/v4"
-	"log"
+	"github.com/labstack/echo/v4/middleware"
+	"go.uber.org/zap"
+	"os"
+	"time"
 )
 
+var sugar zap.SugaredLogger
+
 func main() {
 
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		panic(err)
+	}
+	defer func(logger *zap.Logger) {
+		err := logger.Sync()
+		if err != nil {
+			panic(err)
+		}
+	}(logger)
+
+	sugar = *logger.Sugar()
+
 	cfg, err := config.LoadServerConfig()
 	if err != nil {
-		log.Fatalf("couldn't create a config %s", err)
+		sugar.Fatalf("couldn't create a config %s", err)
 	}
 
 	// from cfg:
-	log.Println("cfg:")
-	log.Println("address:", cfg.Address)
+	sugar.Info(
+		"cfg: \n", cfg.String(),
+	)
+	sugar.Infow(
+		"Starting server",
+	)
+	_, err = os.Stat(cfg.FileStoragePath)
+	if os.IsNotExist(err) {
+		cfg.Restore = false
+		fmt.Printf("%v file does not exist\n", cfg.FileStoragePath)
+	}
+	repository.SetupFilePathStorage(cfg.FileStoragePath)
+	if cfg.Restore && cfg.FileStoragePath != "" {
+		if err = repository.Load(); err != nil {
+			sugar.Fatalf("couldn't restore metrics by %s", err)
+		}
+	}
+
+	if cfg.FileStoragePath != "" {
+		if cfg.StoreInterval != 0 {
+			handler.SyncSaveUpdateHandlerJSON = false
+			ticker := time.NewTicker(time.Duration(cfg.StoreInterval) * time.Second)
+			defer ticker.Stop()
+			go func() {
+				for range ticker.C {
```

Comment:
`C` is a magic name.

Reply:
I agree with you although I can do nothing with it. It is a [part](https://pkg.go.dev/time@go1.20.4#Ticker) of the Go Standard Library.
```go
// A Ticker holds a channel that delivers “ticks” of a clock
// at intervals.
type Ticker struct {
	C <-chan Time // The channel on which the ticks are delivered.
	r runtimeTimer
}
```

Reply:
Probably this helps
```go
tickerChannel := ticker.C
for range tickerChannel {
``` 
instead of
```go
for range ticker.C {
```

Reply:
done